src/huggingface_hub/file_download.py

```diff
@@ -34,7 +35,7 @@
 
 _PY_VERSION: str = sys.version.split()[0]
 
-if tuple(int(i) for i in _PY_VERSION.split(".")) < (3, 8, 0):
+if packaging.version.Version(_PY_VERSION) < packaging.version.Version("3.8.0"):
```

Comment:
I'm trying to check how to handle this properly with poetry returning invalid versions but their website seems to be down with `ERR_SSL_PROTOCOL_ERROR`

Reply:
> With that raising an `InvalidVersion` error, wouldn't it just crash and make the library unusable for someone in the position of https://github.com/huggingface/huggingface_hub/pull/27?

Yes it would exclude "3.9.0+". However, the packaging package claims to implement PEP 440. And according to PEP 440 "3.9.0+" is not a valid version (due to the "+" suffix).

The previous regex proposal could process "3.9.0anything123too".

Reply:
> I'm trying to check how to handle this properly with poetry returning invalid versions but their website seems to be down with `ERR_SSL_PROTOCOL_ERROR`

yesterday pypi.org had an CDN outage no pip install)

Reply:
I could add `.rstrip("+")` as done in https://github.com/huggingface/huggingface_hub/pull/27/files
i.e. `_PY_VERSION: str = sys.version.split()[0].rstrip("+")`
It seems that @ericricky isn't active rn.